src/display.rs

```diff
@@ -587,6 +592,8 @@ mod tests {
             let dummy = Dummy([42; 32]);
             assert_eq!(dummy.to_string(), "2a".repeat(32));
             assert_eq!(format!("{:.10}", dummy), "2a".repeat(5));
+            assert_eq!(format!("{:.11}", dummy), "2a".repeat(5) + "2");
```

Comment:
TIL, I did not know that `String` implemented `ops::Add`.

Reply:
Yes, and I hate it. `+` is commutative in basically every single mathematical context. I complained about this in the pre-1.0 days and said that (a) we should have a `++` operator for concatenation and was told that "this isn't Haskell", and (b) some muddied argument along the lines that `*` is non-commutative for matrices and nobody is bothered by that. IOW "fuck off nerd".

Anyway this is fine in test code but if we were using `+` on strings or vectors in real code I would have NACKed it.

Reply:
Daaamn! This annoyed me too and I was thinking this is one of the few things PHP got right.